api/provider/aws/load_balancer_delete.go

```diff
@@ -44,25 +50,89 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 		}
 	}
 
-	if err := l.deleteTags(loadBalancerID); err != nil {
+	return l.deleteTags(loadBalancerID)
+}
+
+func (l *LoadBalancerProvider) deleteLoadBalancer(loadBalancerName string) error {
+	lb, err := describeLoadBalancer(l.AWS.ELB, l.AWS.ALB, loadBalancerName)
+	if err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == alb.ErrCodeLoadBalancerNotFoundException {
+			return nil
+		}
+
 		return err
 	}
 
+	if lb.isCLB {
```

Comment:
all of these switches seem to indicate a problem with the initial design; I think a better approach here is to have both ALBs and CLBs implement a common interface, and have unique types for both ALB and CLB that know how to Read(), Delete(), Create(), etc. themselves.

Reply:
I would've liked to have done that, but a common interface wouldn't be possible due to the types being in different packages. 

My initial design embedded an ALB & CLB in a common load balancer. But this gets tricky when accessing properties; as you don't know which property belongs to an ALB or a CLB.

Reply:
I'm open to a different approach though for sure.